Core/include/Acts/Navigation/NavigationStateUpdaters.hpp

```diff
@@ -39,31 +39,25 @@ inline void updateCandidates(const GeometryContext& gctx,
   const auto& position = nState.position;
   const auto& direction = nState.direction;
 
-  NavigationState::SurfaceCandidates nextSurfaceCandidates;
-
-  for (NavigationState::SurfaceCandidate c : nState.surfaceCandidates) {
+  for (NavigationState::SurfaceCandidate& c : nState.surfaceCandidates) {
```

Comment:
I think we need some kind of `intersection candidate` model to correctly handle cylinders